web_client/analysis-modules/feature-matching.jsx

```diff
@@ -0,0 +1,170 @@
+
+import actions from '../actions';
+import analysisUtils from '../utils/analysis';
+import { Promise } from '../utils/promise';
+import { store, rest } from '../globals';
+
+import { formatSize } from 'girder/misc';
+
+const D = store.dispatch.bind(store);
+
+let page1;
+let computeResultElement;
+let applyResultElements;
+let tabGroupElement;
+let computeTabElement;
+let applyTabElement;
+let matchChooserElement;
+
+const computeMatch = (data, page) => {
+  const truncatePromise = Promise.all([
```

Comment:
I really can't say that I follow you, here.  What we're looking at is just an array literal (no "'list construction' syntax"), with three elements, formatted in an attempt to make clear where the expression for each element begins and ends.  Clearly, my formatting attempt wasn't entirely successful, but without actionable feedback, I struggle to identify anything concrete worth changing.  The only thing I can think of might be to add an extra level of indentation just for the square brackets to make them stand out more?  I'm really grasping at straws, here.

I think it's also worth pointing out that for function calls with a single array literal argument, I've usually taken the liberty of keeping the array brackets and parentheses together, as in:

    myFunction([
        my,
        array,
        literal
    ]);

And in particular, this comes up a _lot_ with `Promise.all()`, so consider that any recommendations you make here might be applicable to _many_ parts of the code spread out through the whole code base.